ql/src/java/org/apache/hadoop/hive/ql/io/orc/encoded/EncodedTreeReaderFactory.java

```diff
@@ -224,7 +237,252 @@ private static void skipCompressedIndex(boolean isCompressed, PositionProvider i
     index.getNext();
   }
 
-  protected static class StringStreamReader extends StringTreeReader
+  public static class StringDictionaryTreeReaderHive extends TreeReader {
```

Comment:
Why is this added in scope of upgrade?

Reply:
Ok, that seems to be an improvement or say a bug fix in the ORC project and we are just implementing our own varient because now the original class is causing test failures.
This isn't the ideal approach and will backfire in future when we try to upgrade and the changes in ORC depends on the ones which we ditched.

We should try to adapt to those changes and make sure we don't crash with those changes in Hive by making hive changes, rather than maintaining a old version of ORC class at Hive

Reply:
Hi @ayushtkn, I agree with you. It is not ideal approach. Before implementing this approach I did try to adapt Hive, but I didn't succeed to find how Hive could be adapted to ORC-1060 changes because those changes are only inside internal implementation of Orc StringDictionaryTreeReader class. 

I agree with you that this approach is not ideal and will backfire in future when we try to upgrade and the changes in ORC depends on the ones which we ditched, but Hive already heavily depends on internal ORC API by implementing its own column readers on top of ORC and when upgrading to different ORC version it is often required to make adaptations in Hive.

Reply:
@abstractdog any pointers here? Does that sound fine thing to do for now

Reply:
makes sense to me, thanx Laszlo!!!